theSrc/R/htmlwidget.R

```diff
@@ -174,142 +172,127 @@ PalmTrees <- function(
         stop("Input data must be a matrix or a data frame.")
     }
 
-    raw.data = data
-    if (sum(is.na(data)) > 0) {
-        data[is.na(data)] = 0
-        warning("Missing values detected.")
-    }
+    raw.data <- data
 
-    if (sum(data < 0) > 0) {
+    if (any(data < 0, na.rm = TRUE))
         stop("Input data must not contain negative numbers.")
-    }
 
-    nr = nrow(data);
-    nc = ncol(data);
+    nr <- nrow(data)
+    nc <- ncol(data)
 
     if (is.null(weights))
-        weights = rep(1/nc,nc)
+        weights <- rep(1/nc,nc)
     else {
         if (length(weights) != nc) {
             stop("The length of weights must be the same as the number of columns.")
         }
-        weights = weights/sum(weights)
+        weights <- weights/sum(weights)
     }
 
-
     if (is.null(col.names))
-        col.names = colnames(data)
+        col.names <- gsub("(\\r\\n|\\n|\\r)", " ", colnames(data))
```

Comment:
Shouldn't the gsub need to be applied even if !is.null(col.names)?

Reply:
Yes, thanks I've fixed this